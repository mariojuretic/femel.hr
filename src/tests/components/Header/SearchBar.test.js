import React from 'react';
import { shallow } from 'enzyme';
import HeaderSearchBar from '../../../components/Header/SearchBar';

test('should render SearchBar correctly', () => {
  const wrapper = shallow(<HeaderSearchBar />);
  expect(wrapper).toMatchSnapshot();
});

test('should set search state on input change', () => {
  const value = '123abc';
  const wrapper = shallow(<HeaderSearchBar />);
  wrapper.find('input').simulate('change', {
    target: { value }
  });
  expect(wrapper.state('search')).toBe(value);
});