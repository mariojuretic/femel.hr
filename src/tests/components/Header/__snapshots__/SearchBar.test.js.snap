// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBar correctly 1`] = `
<form
  className="header__search"
  onSubmit={[Function]}
>
  <input
    className="header__search-input"
    name="search"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
  />
  <button
    className="header__search-button"
  >
    <i
      className="fas fa-search"
    />
  </button>
  <div
    className="header__search-results"
  >
    <p>
      No results for '
      '
    </p>
  </div>
</form>
`;
